src/Json/Decode.elm

```diff
@@ -309,11 +306,16 @@ null =
     Native.Json.decodeNull
 
 
-{-| Great for handling optional fields. The following code decodes JSON
-objects that may not have a profession field.
+{-| Extract a Maybe value, wrapping successes with `Just` and turning any
+failure in `Nothing`. Great for handling fields that may be missing or null. The
```

Comment:
This is actually not that great for things that may be `null` because it actually admits all sorts of errors. A decoder that can be null or something, is different than a decoder that is something or anything. If the given decoder messes up, it'll give `Nothing` (corresponding to `null` which it is not) rather than failing.